### PR TITLE
feat(demo): implement IntrospectionFragmentMatcher

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
- Adds an architect task for generating fragment types in demo
- Migrates from using `ApolloBoost` to base `Apollo`
- Creates and uses `IntrospectionFragmentMatcher` in `Apollo` config

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -8,6 +8,19 @@
...
       "projectType": "application",
       "architect": {
+        "generateFragmentTypes": {
+          "builder": "@daffodil/builders:generateFragmentTypes",
+          "options": {
+            "url": "",
+            "path": ""
+          },
+          "configurations": {
+            "magento": {
```
Comment (damienwebdev) [inline angular.json:18]: `magento-local`

Comment (griest024) [reply angular.json]: changed

**Comment by griest024** [comment]
@damienwebdev pipelines are good now

**Pull request merged**